internal/graph/track_check_resolver.go

```diff
@@ -0,0 +1,247 @@
+	return func(t *TrackerCheckResolver) {
+		t.logger = logger
+	}
+}
+
+func WithTrackerContext(ctx context.Context) TrackerCheckResolverOpt {
+	return func(t *TrackerCheckResolver) {
+		t.ctx = ctx
+	}
+}
+
+// NewTrackCheckResolver creates an instance tracker Resolver.
+func NewTrackCheckResolver(opts ...TrackerCheckResolverOpt) *TrackerCheckResolver {
+	t := &TrackerCheckResolver{
+		limiter: rate.NewLimiter(rate.Every(trackerLogInterval), trackerLogLines),
+		ticker:  time.NewTicker(randomLogging()),
+	}
+
+	for _, opt := range opts {
+		opt(t)
+	}
+
+	t.delegate = t
+
+	if t.running = t.validate(); t.running {
+		t.launchFlush()
+	}
+
+	return t
+}
+
+// RandomLogging allows multiple processes to log at different interval.
+func randomLogging() time.Duration {
+	val := rand.Intn(trackerMaxInterval-trackerMinInterval+1) + trackerMinInterval
+	trackerInterval = time.Duration(val) * time.Second
+	return trackerInterval
+}
+
+// Validate options.
+func (t *TrackerCheckResolver) validate() bool {
+	if t.ctx == nil {
+		return false
+	}
+
+	if t.logger == nil {
+		return false
+	}
+	return true
+}
+
+// LogExecutionPaths reports the model and tuple path.
+func (t *TrackerCheckResolver) logExecutionPaths(flush bool) {
+	t.nodes.Range(func(k, v any) bool {
+		paths := v.(*sync.Map)
+		storeModel := k.(trackerKey)
+		paths.Range(func(k, v any) bool {
+			path := k.(string)
+			tree := v.(*resolutionNode)
+			if tree.expired() || flush {
+				if !t.limiter.Allow() && !flush {
+					return false
+				}
+				t.logger.Info("execution path hits",
+					zap.String("store", storeModel.store),
+					zap.String("model", storeModel.model),
+					zap.String("path", path),
+					zap.Uint64("hits", tree.hits.Load()))
+
+				paths.Delete(path)
+			}
+			return true
+		})
+		return true
+	})
+}
+
+// LaunchFlush starts the execution path logging and removal of entries.
+func (t *TrackerCheckResolver) launchFlush() {
+	go func() {
+		for {
+			select {
+			case <-t.ctx.Done():
+				t.ticker.Stop()
+				return
+			case <-t.ticker.C:
+				t.logExecutionPaths(false)
+				t.ticker.Reset(trackerInterval)
+			}
+		}
+	}()
+}
+
+// SetDelete assigns the next deletegate in the chain.
+func (t *TrackerCheckResolver) SetDelegate(delegate CheckResolver) {
+	t.delegate = delegate
+}
+
+// GetDelegate return the assigned delegate.
+func (t *TrackerCheckResolver) GetDelegate() CheckResolver {
+	return t.delegate
+}
+
+// Close implements CheckResolver.
+func (t *TrackerCheckResolver) Close() {
+	t.logExecutionPaths(true)
+}
+
+// UserType returns the associated tuple user type.
+func (t *TrackerCheckResolver) userType(userKey string) string {
+	return string(tuple.GetUserTypeFromUser(userKey))
+}
+
+// GetTK returns formatted tuple suitable insertion into list formatted object#relation@user.
+func (t *TrackerCheckResolver) getTK(tk *openfgav1.TupleKey) string {
+	return fmt.Sprintf("%s#%s@%s", tk.GetObject(), tk.GetRelation(), t.userType(tk.GetUser()))
```

Comment:
Do we need to distinguish between the same tuple conditioned vs non-conditioned?